egs/wsj/s5/utils/data/get_allowed_durations.py

```diff
@@ -0,0 +1,194 @@
+#!/usr/bin/env python3
+
+# Copyright     2017  Hossein Hadian
+#               2019  Facebook Inc. (Author: Vimal Manohar)
+# Apache 2.0
+
+
+""" This script generates a set of allowed lengths of utterances
+    spaced by a factor (like 10%). This is useful for generating
+    fixed-length chunks for chain training.
+"""
+
+import argparse
+import os
+import sys
+import copy
+import math
+import logging
+
+sys.path.insert(0, 'steps')
+import libs.common as common_lib
+
+logger = logging.getLogger('libs')
+logger.setLevel(logging.INFO)
+handler = logging.StreamHandler()
+handler.setLevel(logging.INFO)
+formatter = logging.Formatter("%(asctime)s [%(pathname)s:%(lineno)s - "
+                              "%(funcName)s - %(levelname)s ] %(message)s")
+handler.setFormatter(formatter)
+logger.addHandler(handler)
+
+def get_args():
+    parser = argparse.ArgumentParser(description="""
+    This script creates a list of allowed durations of
+    utterances for flatstart LF-MMI training
+    corresponding to input data directory 'data_dir'
+    and writes it to allowed_durations.txt file in the
```

Comment:
This usage message does not very clearly explain what the script does.  It would be nice to know, at the very least, what the output looks like, e.g. what the format is..

Reply:
I fixed the usage message. Ideally utils/data/perturb_speed_to_allowed_lengths.py. could be modified to use this. But I didn't want to affect other scripts without extensively testing it.